libp2p/Host.cpp

```diff
@@ -494,6 +503,12 @@ void Host::registerCapability(std::shared_ptr<CapabilityFace> const& _cap)
 void Host::registerCapability(
     std::shared_ptr<CapabilityFace> const& _cap, std::string const& _name, u256 const& _version)
 {
+    if (haveNetwork())
+    {
+        cwarn << "Capabilities must be registered before the network is started";
```

Comment:
Why this new requirement appeared?

Reply:
Because we determine whether or not we start the TCP listener based on whether or not there are registered capabilities, and we start the TCP listener in network thread startup (`host::startedWorking`), meaning capabilities need to be registered before network start. I suppose I could add functionality which detects when capabilities are registered after network start and starts the TCP listener, but this would make the changes more complicated and I didn't think it was necessary because I went through the RPC functions and didn't see any which allow someone to register capabilities.